File: data/golang/role_login.go
package pb

import (
	"packet"
)

// 请求登录
type RoleLogin struct {
	Account                  string //账号
	Password                 string //密码
}

func (this *RoleLogin) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteString(this.Account)
	pack.WriteString(this.Password)

	return pack.Encode(uint16(1010))
}

func (this *RoleLogin) EncodeMsg() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteString(this.Account)
	pack.WriteString(this.Password)

	return pack.ReadBytes()
}

func RoleLoginDecode(pack *packet.Packet) *RoleLogin {
	roleLogin := &RoleLogin{}

	roleLogin.Account = pack.ReadString()
	roleLogin.Password = pack.ReadString()
	return roleLogin
}

File: data/typescript.cc/RoleLogin.ts
import Packet from '../net/Packet';


// 物品数据
export default class RoleLogin {
	private _account: string = ""; //账号
	private _password: string = ""; //密码

	constructor(packet?: Packet) {
		if (packet) {
			this._account = packet.ReadString();
			this._password = packet.ReadString();
		}
	}

	public Encode(): Packet {
		const packet = this._encode();
		packet.Encode(1010);
		return packet;
	}

	public GetBuffer(): ByteBuffer {
		return this._encode().GetBuffer();
	}

	private _encode(): Packet {
		let packet: Packet = new Packet();
		packet.WriteString(this._account);
		packet.WriteString(this._password);
		return packet;
	}

	public get account(): string { return this._account; }
	public set account(value: string) { this._account = value; }
	public get password(): string { return this._password; }
	public set password(value: string) { this._password = value; }
}
